### PR TITLE
[Optimize]Take all scan nodes of one sql into consideration when select host for a tablet

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

Currently when a scan node scans many tablets, Doris will assure it load balance when choosing which replica for scan task to be executed. But it does not take other scan nodes into consideration to implement a global load balance. This patch tries to make all tables of all scan nodes to be load balance.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by caiconghui** [comment]
@xinghuayu007 please resolve conflict firstly

**Comment by xinghuayu007** [comment]
> @xinghuayu007 please resolve conflict firstly

done

**Review by caiconghui — DISMISSED** [review]
+1

File: fe/fe-core/src/test/java/org/apache/doris/qe/CoordinatorTest.java [hunk 1]
```diff
@@ -595,5 +597,137 @@ public void testBucketShuffleWithUnaliveBackend()  {
...
+        ImmutableMap<Long, Backend> idToBackend =
+                new ImmutableMap.Builder<Long, Backend>().
+                        put(0l, backend0).
+                        put(1l, backend1).
+                        put(2l, backend2).build();
+        Deencapsulation.setField(coordinator, "idToBackend", idToBackend);
+
+        Deencapsulation.invoke(coordinator, "computeScanRangeAssignment");
+        FragmentScanRangeAssignment assignment = fragmentExecParamsMap.get(fragment.getFragmentId()).scanRangeAssignment;
+        for (Map.Entry<TNetworkAddress, Map<Integer, List<TScanRangeParams>>> entry:assignment.entrySet()) {
```
Comment (caiconghui) [inline fe/fe-core/src/test/java/org/apache/doris/qe/CoordinatorTest.java:715]:
```suggestion
        for (Map.Entry<TNetworkAddress, Map<Integer, List<TScanRangeParams>>> entry : assignment.entrySet()) {
```

Comment (caiconghui) [reply fe/fe-core/src/test/java/org/apache/doris/qe/CoordinatorTest.java:715]: and the same as following

Comment (xinghuayu007) [reply fe/fe-core/src/test/java/org/apache/doris/qe/CoordinatorTest.java]: done

**Review by caiconghui — APPROVED** [review]
+1

**Pull request merged**